therapy/query.py

```diff
@@ -442,15 +484,38 @@ def _handle_failed_merge_ref(self, record: Dict, response: Dict,
         response["match_type"] = MatchType.NO_MATCH
         return response
 
-    def search_groups(self, query: str) -> Dict:
+    def _resolve_merge(self, response: Dict, query: str,
+                       record: Dict, match_type: MatchType) -> Dict:
+        """Given a record, return the corresponding normalized record
+
+        :param Dict response: in-progress response object
+        :param str query: exact query as provided by user
+        :param Dict record: TODO
+        :param MatchType match_type: type of match that returned these records
+        :return: formed response, a Dict
+        """
+        merge_ref = record.get("merge_ref")
+        if merge_ref:
+            # follow merge_ref
+            merge = self.db.get_record_by_id(merge_ref, False, True)
+            if merge is None:
+                return self._handle_failed_merge_ref(record, response, query)
+            else:
+                return self._add_vod(response, merge, query, match_type)
+        else:
+            # record is sole member of concept group
+            return self._add_vod(response, record, query, match_type)
+
+    def search_groups(self, query: str, infer: bool = True) -> Dict:
         """Return merged, normalized concept for given search term.
 
         :param str query: string to search against
+        :param bool infer: if true, try to infer namespace for IDs
         """
         # prepare basic response
         response: Dict[str, Any] = {
             "query": query,
-            "warnings": self._emit_warnings(query),
+            "warnings": self._emit_char_warnings(query),
```

Comment:
This wasnt changed in this PR, but you don't need to set the url in ServiceMeta. Can change here or in a different issue

Reply:
e z fix